evil_mystery_word.py

```diff
@@ -0,0 +1,252 @@
+# Started attempting this on Saturday night. Looked at a couple of other
+# peoples attempts at it (people not in The Iron Yard). Didn't exactly use
+# what they did, but got many of the basic ideas which led to my solution.
+
+
+import random
+from pyfiglet import Figlet
+import os
+import time
+
+
+f = Figlet(font='epic')
+
+
+def list_maker(recent_list, recent_guess):
+    """Takes the largest list to date and the most recent user input. Then
+    makes a dictionary which has a key of each pattern and the value is a
+    list of words with the matching pattern. At the end it checks to see which
+    pattern has the largest number of words and that becomes the new largest
+    list."""
+    dict_of_patterns = {}
+    for word in recent_list:
+        word = word.upper()
+        new_pattern = []
+        for letter in word:
+            if letter == recent_guess:
+                new_pattern.append(letter)
+            else:
+                new_pattern.append("_")
+        pattern = "".join(new_pattern)
+        if pattern not in dict_of_patterns:
+            dict_of_patterns[pattern] = [word]
+        else:
+            dict_of_patterns[pattern].append(word)
+    largest_list = dict_of_patterns[max(dict_of_patterns)]
+    return largest_list
+
+
+def change_guess_counter(guess_counter, guessed_letter, game_word):
```

Comment:
This is a great function. Doing this logic in a function instead of conditionally deciding whether to call the function is a nice move.